app/namespaces/test_adjustment/test_adjustment_service.py

```diff
@@ -0,0 +1,267 @@
+            adapt_delta = 0.8
+        fit = self.TESTADJ_MODEL.sampling(data=model_params,
+                                          iter=self.n_iter,
+                                          chains=self.n_chains,
+                                          control={'adapt_delta': adapt_delta},
+                                          check_hmc_diagnostics=False)
+
+        summary = fit.summary()
+        summary_df = pd.DataFrame(data=summary['summary'],
+                                  index=summary['summary_rownames'],
+                                  columns=summary['summary_colnames'])
+
+        samples = fit.extract(pars='prev', permuted=True)['prev']
+
+        summary_df_parsed = summary_df.loc['prev', ['50%', '2.5%', '97.5%']]
+        summary_df_parsed['samples'] = samples
+        summary_df_parsed['fit'] = fit
+
+        diagnostics = pystan.diagnostics.check_hmc_diagnostics(fit)
+
+        satisfactory_model_found = all(diagnostics.values())
+        return summary_df_parsed, satisfactory_model_found
+
+    def pystan_adjust(self, model_params: Dict, execution_params: Dict = {}) -> Union[Tuple, pystan.StanModel]:
+        if self.n_replicates % 2 != 1:
+            raise ValueError(f'n_replicates must be odd')
+        credible_interval_size = execution_params.get('credible_interval_size', 0.95)
+
+        # Validate model_params using marshmallow
+        try:
+            ModelParamsSchema().load(model_params)
+        except ValidationError as err:
+            print("Error: ", err.messages)
+            return None, None, None
+
+        satisfactory_model_set_found = False
+        n_modelsets = 0
+
+        while not satisfactory_model_set_found:
+            satisfactory_model_set = []
+            for _ in range(self.n_replicates):
+                n_trials = 0
+                satisfactory_model_found = False
+
+                while not satisfactory_model_found:
+                    # Attempt to fit a model
+                    summary_df_parsed, satisfactory_model_found = self.fit_one_pystan_model(model_params)
+                    # If number of attempts exceeded, return Nones
+                    n_trials += 1
+                    if n_trials >= self.trials_lim:
+                        print('no models met the HMC diagnostics in {trials_lim} trials')
+                        return None, None, None
+
+                satisfactory_model_set.append(summary_df_parsed)
+
+            satisfactory_models_df = pd.DataFrame(satisfactory_model_set)
+
+            # see whether all results are within 10% of median; if not, rerun
+            model_medians = satisfactory_models_df['50%']
+            median = model_medians.median()
+            consistent_results_found = model_medians.between(0.9 * median, 1.1 * median).all()
+            median_run = satisfactory_models_df[satisfactory_models_df['50%'] == median].iloc[0]
+
+            lower, upper = arviz.hdi(median_run['samples'], credible_interval_size)
```

Comment:
Need to make sure to keep lines 162-169 - we still need to calculate the confidence interval and check whether the result is bounded